modin/pandas/accessor.py

```diff
@@ -0,0 +1,85 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pandas
+from pandas.core.arrays.sparse.dtype import SparseDtype
+
+
+class BaseSparseAccessor:
+    _validation_msg = "Can only use the '.sparse' accessor with Sparse data."
+
+    def __init__(self, data=None):
+        self._parent = data
+        self._validate(data)
+
+    def _validate(self, data):
+        raise NotImplementedError
```

Comment:
if we suppose to not implementing this in a base class, we should mark it as `@abc.abstract`

Reply:
We mark methods as `@abc.abstract` which have default implementation like `pass`. However, this method has implementation like [this](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/axis_partition.py#L75), for instance, which differs from default implementation.

Reply:
I don't get it.

`BaseSparseAccessor` seems to be a class, which has a method that **have to be** implemented in children classes, and we can't create an instance of it without implementation of `_validate` method (because we're calling it inside constructor). So going by this description `BaseSparseAccessor` considered to be an abstract, `_validate` considered to be an abstract method, and should be marked as `@abc.abstract`?

Another point:
If we'll try to create an instance of `BaseSparceAccessor` we will get `NotImplementedError`, and it could not clear what's going on. But if the method will be marked as abstract, that we will get an error that we can't create an instance of abstract classes, it's seems to be a more meaningful error message.

Reply:
If we mark `BaseSparseAccessor` as `abc`, it will strange because the class contains `__init__` method. I took the logic from pandas and would keep it as is.

Reply:
ok